apis/api_extension/v1alpha1/visibility_types.go

```diff
@@ -0,0 +1,48 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +kubebuilder:object:root=true
+// +k8s:openapi-gen=true
+
+// User-facing representation of a pending workload that summarizes the relevant information for
+// position in the cluster queue.
+type PendingWorkloadSummary struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+// +k8s:openapi-gen=true
+
+// PendingWorkloadSummaryList contains a list of pending workloads in the context
+// of the query (within LocalQueue or ClusterQueue).
+type PendingWorkloadSummaryList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+
+	Items []PendingWorkloadSummary `json:"items"`
+}
+
+func init() {
+	SchemeBuilder.Register(&PendingWorkloadSummary{}, &PendingWorkloadSummaryList{})
```

Comment:
This looks odd compared to https://github.com/kubernetes-sigs/kueue/blob/0adc8c45dabaf08eb72bc27877515841eaf0a1c4/apis/kueue/v1beta1/workload_types.go#L300

In there, WorkloadList is used to list workloads, when the URL refers to workloads.

In this case, the URL refers to clusterqueue, but the elements are workloads?

Reply:
I've changed the URLs